tensorflow/python/ops/bincount_ops.py

```diff
@@ -134,7 +134,7 @@ def bincount(arr,
         weights = ops.convert_to_tensor(weights, name="weights")
         return gen_math_ops.unsorted_segment_sum(weights, arr, output_size)
       weights = constant_op.constant([], dtype)
-      return gen_math_ops.bincount(arr, output_size, weights)
+      #return gen_math_ops.bincount(arr, output_size, weights)
```

Comment:
@cheshire I've temporary commented this as the python logic, also reading the official documentation of the op, it seems not clear.

I don't understand what is the role of this kernel, it is not covered by the python ops tests, and it is also failing with non dense inputs:
https://colab.research.google.com/gist/bhack/7158a476d944719825709c69ad9f7c32/untitled139.ipynb

Reply:
It seems that their is a specialized TF kernel for deterministic GPU.
The python logic it still seems strange as it doesn't check the tensor input type and so it is automatically failing with non dense input tensors.

Reply:
I've pushed something to pass the tests minimizing the impact on the python op and to not duplicate code in TF2XLA.

I still think that we have a bug in the python op logic as it doesn't check the tensor input type in this specific case. 
I have opened a new ticket for this at:
https://github.com/tensorflow/tensorflow/issues/56499